renderer/FastBufferDump.cpp

```diff
@@ -25,40 +25,20 @@ FastBufferDump::FastBufferDump(uint32_t num, uint32_t z, buffer_handle_t b, int
 }
 
 void FastBufferDump::prepareFolder() {
-    struct stat st;
-    bool exist = false;
-
-    mEnableDump = false;
-    int ret = stat(kDumpFolder, &st);
-    if (ret < 0) {
-        if (errno != ENOENT) {
-            ALOGE("Failed to access %s:%s", kDumpFolder, strerror(errno));
-            return;
-        }
-    } else {
-        if (!(st.st_mode & S_IFDIR)) {  // not a folder
-            ALOGD("%s exits but it is not a folder, try to recreate it as folder", kDumpFolder);
-            if (unlink(kDumpFolder) < 0) {
-                ALOGE("Can't remove %s:%s", kDumpFolder, strerror(errno));
-                return;
-            }
-        } else {
-            exist = true;
-        }
+    int ret = unlink(kDumpFolder);
+    if (ret != 0 && ret != -ENOENT) {
+        ALOGE("Can't remove %s:%s", kDumpFolder, strerror(errno));
+        return;
     }
-    // now create the folder
-    if (!exist) {
-        if (mkdir(kDumpFolder, 0777) < 0) {
-            ALOGE("%s doesn't exit but failed to create it:%s", kDumpFolder, strerror(errno));
-            return;
-        }
+    if (mkdir(kDumpFolder, 0777) < 0) {
```

Comment:
no need to check to return since it will fail if the folder exists. We just need to make sure the folder is accessible in next access() check.

Reply:
I'm afraid that the coverity scan will report this a issue if it doesn't check mkdir return. I will remove unlink and keep the mkdir return check.